scylla.go

```diff
@@ -174,14 +201,24 @@ func isScyllaConn(conn *Conn) bool {
 // connections to already opened shards. Keeping excess connections open helps
 // reaching equilibrium faster since the likelihood of hitting the same shard
 // decreases with the number of connections to the shard.
+//
+// scyllaConnPicker keeps track of the details about the shard-aware port.
+// When used as a Dialer, it connects to the shard-aware port instead of the
+// regular port (if the node supports it). For each subsequent connection
+// it tries to make, the shard that it aims to connect to is chosen
+// in a round-robin fashion.
 type scyllaConnPicker struct {
-	address     string
-	conns       []*Conn
-	excessConns []*Conn
-	nrConns     int
-	nrShards    int
-	msbIgnore   uint64
-	pos         uint64
+	address                string
+	shardAwareAddress      string
+	conns                  []*Conn
+	excessConns            []*Conn
+	nrConns                int
+	nrShards               int
+	msbIgnore              uint64
+	pos                    uint64
+	dialer                 Dialer
```

Comment:
I'm thinking if it belong here? Maybe a new type would be better? Maybe they can be linked with a pointer. What is the benefit of exposing scyllaConnPicker as a dialer? Note that scyllaConnPicker is not thread safe so I'd prefer to keep it's usage restricted.

Reply:
I see `dialer, ok := pool.connPicker.(Dialer)` this is fine but maybe we can embed another type that would implement the dialer functionality and keep them separated.

Reply:
I reorganized it a little:
- `ConnPicker` gains a `GetCustomDialer` method,
- `scyllaConnPicker.GetCustomDialer` chooses a free shard and returns a `scyllaOneShardDialer` which calls only that shard.